File: 1189-maximum-number-of-balloons/1189-maximum-number-of-balloons.cpp
class Solution {
public:
    int maxNumberOfBalloons(string text) {
        int bCnt = 0, aCnt = 0, lCnt = 0, oCnt = 0, nCnt = 0;
        for (int i = 0; i < text.size(); ++i){
            if (text[i] == 'b') ++bCnt;
            else if (text[i] == 'a') ++aCnt;
            else if (text[i] == 'l') ++lCnt;
            else if (text[i] == 'o') ++oCnt;
            else if (text[i] == 'n') ++nCnt;
        }
        lCnt /= 2;
        oCnt /= 2;
        
        return min({bCnt, aCnt, lCnt, oCnt, nCnt});
        
    }
};

File: 1189-maximum-number-of-balloons/1189-maximum-number-of-balloons.py
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        bCnt, aCnt, lCnt, oCnt, nCnt = 0, 0, 0, 0, 0
        
        for c in text:
            if c == 'b':
                bCnt += 1
            elif c == 'a':
                aCnt += 1
            elif c == 'l':
                lCnt += 1
            elif c == 'o':
                oCnt += 1
            elif c == 'n':
                nCnt += 1
        
        lCnt //= 2
        oCnt //= 2
        
        return min([bCnt, aCnt, lCnt, oCnt, nCnt])